access/src/main/java/org/apache/accumulo/access/AccessEvaluator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.access;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * <p>
+ * An implementation of the Accumulo visibility standard as specified in this document (TODO write
+ * the document based on current Accumulo implementation and post somewhere).
+ *
+ * <p>
+ * Below is an example that should print false and then print true.
+ *
+ * <pre>
+ * {@code
+ * var evaluator = VisibilityArbiter.builder().authorizations("ALPHA", "OMEGA").build();
+ *
+ * System.out.println(evaluator.isAccessible("ALPHA&BETA"));
+ * System.out.println(evaluator.isAccessible("(ALPHA|BETA)&(OMEGA|EPSILON)"));
+ * }
+ * </pre>
+ *
+ *
+ * @since ???
+ */
+public interface AccessEvaluator {
```

Comment:
I think that users may to want to create an Authorizations object for reuse. ALso, I think we should consider supporting multiple Authorizations in the call to the evaluator. If you think about a chain of entities being involved in the scan, we could evaluate that each one can access the data.

Reply:
We do need to support the use case of multiple sets of authorizations somehow.

Reply:
What do you think about something like:
```
Attributes user = Attributes.build("ALPHA", "OMEGA");
Attributes server = Attributes.build("ALPHA", "BETA");

AccessEvaluator evaluator = AccessEvaluator.build(Set.of(user, server));

Scanner.forEach((k,v) -> {
  System.out.println(evaluator.canAccess(k.getColumnVisibility().getBytes());
});
```

Reply:
I added support for multiple authorization sets in 763c4a8 in the implementation.  Still need to work on improving the API for multiple sets of authorizations.

Reply:
Updated API for multiple auth sets in b0de391c390bc471446c71d137616a4b9edd4f84.  Also made some API updates unrelated to multiple auths set.